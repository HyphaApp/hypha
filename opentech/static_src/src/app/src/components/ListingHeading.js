import React from 'react';
import PropTypes from 'prop-types';

export default class ListingHeading extends React.Component {
    render() {
        return (
            <li className="listing__item listing__item--heading">
                <h5 className="listing__title">{this.props.title}</h5>
                <span className="listing__count">{this.props.count}</span>
            </li>
        );
    }
}

ListingHeading.propTypes = {
    title: PropTypes.string,
    count: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
};
