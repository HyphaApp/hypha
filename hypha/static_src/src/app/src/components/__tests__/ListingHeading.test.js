import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import ListingHeading from "../ListingHeading";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Listing Heading component", () => {
    const title = "Title Text"
    const count = "1"
    const subject = mount(<ListingHeading title={title} count={count}/>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.listing__item--heading').length).toBe(1);
      expect(subject.find('.listing__item--heading').props().id).toEqual("title-text");

      expect(subject.find('h5').text()).toBe(title);
      expect(subject.find('div').children().length).toBe(2);
      expect(subject.find('span').text()).toBe(count);
    });
  
    test('render a listing heading component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
