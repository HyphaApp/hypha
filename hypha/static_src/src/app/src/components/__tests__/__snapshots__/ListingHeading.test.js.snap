// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Listing Heading component render a listing heading component 1`] = `
<ListingHeading
  count="1"
  title="Title Text"
>
  <div
    className="listing__item listing__item--heading"
    id="title-text"
  >
    <h5
      className="listing__title"
    >
      Title Text
    </h5>
    <span
      className="listing__count"
    >
      1
    </span>
  </div>
</ListingHeading>
`;
